src/auto-router/auto-router.ts

```diff
@@ -218,127 +239,24 @@ export class AutoRouter implements Router {
-      buyAmounts.map(async ({ asset, buyAssetMinAmount }, amountIndex) => {
-        const price =
-          await priceOracle.contract.callStatic.refreshedAssetPerBaseInUQ(asset)
-        return {
-          asset,
-          buyAmount: BigNumber.from(buyAssetMinAmount)
-            .mul(BigNumber.from(2).pow(112))
-            .mul(255)
-            .div(price.mul(amounts[amountIndex]!.weight)),
-        }
-      }),
-    )
-
-    const minAmount = buyAmountsInBase.reduce((min, curr) =>
-      min.buyAmount.lte(curr.buyAmount) ? min : curr,
-    )
-
-    const scaledSellAmounts = Object.values(amounts).map(({ amount }, i) =>
-      amount.mul(minAmount.buyAmount).div(buyAmountsInBase[i]!.buyAmount),
-    )
-
-    const quotes = await Promise.all(
-      amounts.map(async ({ asset }, i) => {
-        const scaledSellAmount = scaledSellAmounts[i] as BigNumber
-        if (asset === routerInputTokenAddress || scaledSellAmount.isZero())
-          return {
-            asset,
-            swapTarget: constants.AddressZero,
-            buyAssetMinAmount: scaledSellAmounts[i]!,
-            assetQuote: [],
-            estimatedGas: 0,
-          }
-
-        const {
-          buyAmount,
-          to: swapTarget,
-          data: assetQuote,
-          estimatedGas,
-        } = await this.zeroExAggregator.quote(
-          routerInputTokenAddress,
-          asset,
-          scaledSellAmount,
-          options?.zeroExOptions,
-        )
-
-        let buyAssetMinAmount = BigNumber.from(buyAmount)
-        if (options?.zeroExOptions?.slippagePercentage) {
-          buyAssetMinAmount = buyAssetMinAmount
-            .mul(1000 - options?.zeroExOptions?.slippagePercentage * 1000)
-            .div(1000)
-        }
-
-        return {
-          asset,
-          buyAssetMinAmount,
-          swapTarget,
-          assetQuote,
-          estimatedGas,
-        }
-      }),
-    )
-
-    amountInInputToken = scaledSellAmounts.reduce(
-      (sum, curr) => sum.add(curr),
-      BigNumber.from(0),
-    )
-
-    const mintOptions = {
-      index: index.address,
-      recipient: await this.indexRouter.account.address(),
-      quotes,
-      amountInInputToken,
-      inputToken: routerInputTokenAddress,
+    if (!inputTokenAddress) {
+      return this.indexRouter.mintSwap(index.address, amountInInputToken)
     }
 
-    return this.indexRouter.mintSwap(
-      mintOptions,
-      amountInInputToken,
+    const { to, buyAmount, data } = await this.zeroExAggregator.quote(
       inputTokenAddress,
-      options?.permitOptions,
+      await this.indexRouter.weth(),
+      amountInInputToken,
     )
+    const params: IReserveRouter.QuoteParamsStruct = {
+      input: inputTokenAddress, //USDC address
+      inputAmount: amountInInputToken, //amount in USDC
+      minOutputAmount: buyAmount, //min amount in weth
+      swapTarget: to, //0x router contract(0xdefi..)
+      assetQuote: data, //qoute(token to WETH)
```

Comment:
if we still need those comments, 
```suggestion
      assetQuote: data, // quote(token to WETH)
```